whitepapers/Dates.md

```diff
@@ -0,0 +1,72 @@
+# Partially Unknown Dates
+
+## Overview
+Many works in SHARE have the wrong publication date displayed, because many sources provide only partial publication dates for some works, such as "May 2016" or just "2016".
+When the day of the month is omitted, the current date parsing logic assumes the first day of the month. When the month is omitted, it assumes January.
+
+### Goals
+Allow storing and displaying works with partially-known publication dates.
+
+## Changes
+
+Create a custom `PartialDate` class with:
+* Three main properties:
+    * `year`: integer (required)
+    * `month`: integer (optional)
+    * `day`: integer (optional)
+* Missing/null values indicate an unknown date part
+* Validation that set values are the correct type and correspond to a valid date
+    * If `month` is missing, `day` must also be missing
+* A way to construct a `PartialDate` from a `Date`-like object, string, or 8-digit integer (see below)
+
+### Models
+* Custom Django field `PartialDateField` that inherits from `IntegerField`
+    * Store a `PartialDate` instance as an 8-digit integer in the database, `YYYYMMDD`
+    * If `month` or `day` is null, put `00` in the `MM` or `DD` segment, respectively
+        * "February 26, 2016" stored as `20160226`
+        * "February 2016" stored as `20160200`
+        * "2016" stored as `20160000`
+    * This allows sorting and filtering by a `PartialDateField` in the database
+        * Values with unknown month and day are sorted at the beginning of the year
+        * Values with known month but unknown day are sorted at the beginning of the month
+* Make `AbstractCreativeWork.date_published` a nullable `PartialDateField`
+
+### Migrations
+* Change the `date_published` columns on `share_creativework` and `share_creativeworkversion` from `DATETIME` to `INTEGER`
```

Comment:
* Create a new nullable `INTEGER` column `date_published_partial`
* Find all special case dates, modify, and copy them over appropriately 
* Copy and truncate all remaining values
* Tell Django that `date_published`'s column is `date_published_partial`
* Add a new field `_date_published` pointing at the column `date_published`
    * So we can keep it around for a little while
* Drop the original `date_published` column in the next release